.github/actions/bump-submodule/action.yml

```diff
@@ -0,0 +1,37 @@
+name: 'Bump submodule'
+description: 'Bump specified third-party submodule for integration testing'
+
+inputs:
+  submodule:
+    description: Name of submodule to bump.
+    required: true
+    type: string
+  revision:
+    description: Git revision from submodule repository
+    required: true
+    type: string
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Bump ${{ inputs.submodule }} submodule
+      shell: bash
+      run: |
+        pushd third_party/${{ inputs.submodule }}
+        git fetch origin
+        git checkout ${{ inputs.revision }}
+        popd
+        # XXX: export the following environment variables to
+        # suppress git config warning. The commit is required to
+        # avoid resetting dirty changes in the submodule.
+        export GIT_COMMITTER_NAME="Integration tester"
```

Comment:
Let's put this in the github environment variables block.
Details:
https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

Reply:
You can't put an `env` section into a GitHub action, even though you can do this for a workflow. In fact, this changeset's second commit fixes exactly that: we are going back to environment variables in script, rather then in a dedicated section.